xdsl/dialects/memref.py

```diff
@@ -304,5 +304,27 @@ def from_memref(cls, memref: Operation | SSAValue):
         return cls.build(operands=[memref], result_types=[IndexType()])
 
 
-MemRef = Dialect([Load, Store, Alloc, Alloca, Dealloc, GetGlobal, Global, Dim],
-                 [MemRefType, UnrankedMemrefType])
+@irdl_op_definition
+class ExtractAlignedPointerAsIndexOp(Operation):
+    name = "memref.extract_aligned_pointer_as_index"
+
+    source: Annotated[Operand, MemRefType]
+
+    aligned_pointer: Annotated[OpResult, IndexType]
+
+    @classmethod
+    def get(cls, source: SSAValue | Operation):
```

Comment:
Why are you using `cls` here? I know it pretty much jsut becomes whatever the callee was, but I am wondering why. I would prefer to not use it and rather use the actual class name (same for the `Rank` and `Dim` constructors, I guess that slipped my review last time).

Reply:
I just checked, it was in the RISC-V dialect, where it made sense because there was a subclass. I'll get that changed!